auxiliary_files/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+# Copyright 2022 The Mumble Developers. All rights reserved.
+# Use of this source code is governed by a BSD-style license
+# that can be found in the LICENSE file at the root of the
+# Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+include(delayed_configure_files)
+
+if(NOT BUILD_RELEASE_DATE)
+	# If BUILD_RELEASE_DATE has not been set, default to time of build
+	string(TIMESTAMP BUILD_RELEASE_DATE "%Y-%m-%d")
+endif()
+
+if(client)
+	if(UNIX AND NOT APPLE)
+		delayed_configure_files(
+			TARGET mumble
+			FILES
+			"${CMAKE_CURRENT_SOURCE_DIR}/config_files/info.mumble.Mumble.appdata.xml.in=${CMAKE_CURRENT_BINARY_DIR}/info.mumble.Mumble.appdata.xml"
+			"${CMAKE_CURRENT_SOURCE_DIR}/config_files/info.mumble.Mumble.desktop.in=${CMAKE_CURRENT_BINARY_DIR}/info.mumble.Mumble.desktop"
+			PASSTHROUGH_VARIABLES
+				MUMBLE_BUILD_YEAR
+				CMAKE_PROJECT_VERSION
+				BUILD_RELEASE_DATE
+				CMAKE_PROJECT_HOMEPAGE_URL
+				CMAKE_PROJECT_DESCRIPTION
+			VARIABLES
+				"MUMBLE_CLIENT_BINARY_NAME=$<TARGET_FILE_NAME:mumble>"
+			PYTHON_INTERPRETER "${PYTHON_INTERPRETER}"
+			@ONLY
+		)
+
+		install(FILES "${CMAKE_BINARY_DIR}/info.mumble.Mumble.appdata.xml" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/metainfo")
+		install(FILES "${CMAKE_BINARY_DIR}/scripts/info.mumble.Mumble.desktop" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications")
+
+		# Install Mumble client man files
+		install(FILES "man_files/mumble.1" DESTINATION "${MUMBLE_INSTALL_MANDIR}" COMPONENT doc)
+
+	endif()
+endif()
+
+if(server)
+	file(COPY "mumble-server.ini" DESTINATION ${CMAKE_BINARY_DIR})
+
+	if(UNIX AND NOT APPLE)
+		delayed_configure_files(
+			TARGET mumble-server
+			FILES
```

Comment:
As I don't know this macro: Are these files installed?

If not, e.g. for the systemd service file you will want to install it to the correct system service location. You can use the output of `pkgconf --variable=systemd_system_unit_dir systemd` (on Arch it is `/usr/lib/systemd/system` but on other distributions it might be elsewhere).

Reply:
No, they are not yet installed.

Reply:
> You can use the output of pkgconf --variable=systemd_system_unit_dir systemd

That would put a dependency on `pkgconf` being installed on the system (e.g. on mine it doesn't seem to be installed) :thinking: 

According to https://unix.stackexchange.com/questions/379411/where-do-i-put-my-systemd-service one is supposed to put the service files under `/etc/systemd/system`. Could you verify that this location would also work on Arch?

Reply:
No, `/etc/systemd/system/` is the location for the system administrator overrides.

When mumble is packaged, its files go to e.g. `/usr/lib/systemd/system/`. I'm writing *e.g.* because on Debian it is `/lib/systemd/system/` and with pkg-config/pkgconf it is straight forward and well supported/understood to get to the correct location of these files.

Reply:
But by using a directory that is different on different distros, we make it unnecessarily complicated to build a package for distro X while running distro Y
(These install paths are the foundation for building packages via CPack)

What issues are created when packages install to the "admin dir"?

Reply:
> But by using a directory that is different on different distros, we make it unnecessarily complicated to build a package for distro X while running distro Y

No, it's not complicated. That's why pkg-config exists! :)

> What issues are created when packages install to the "admin dir"?

Every downstream distribution has to move those files manually if you install them to the wrong location (which is the case with the administrator override location).

Reply:
You can use `pkg-config` (or `pkgconf`, which is a different implementation of it). CMake has had integration for retrieving variables since 3.4: https://cmake.org/cmake/help/latest/module/FindPkgConfig.html#command:pkg_get_variable

Reply:
Meh... Seems like the variable names are different when using `pkg-config` vs `pkgconf`. On my system `pkg-config --variable=systemd_system_unit_dir systemd` returns an empty string, whereas `pkg-config --variable=systemdsystemunitdir systemd` returns `/lib/systemd/system`